File: RepoTransBench/target_projects/Java/C#/happyfish100_fastdfs-client-java/tests/original/DownloadCallbackTest.cs
using Xunit;
using FastDFSClient;

namespace FastDFSClient.Tests
{
    public class DownloadCallbackTest
    {
        private class TestDownloadCallback : IDownloadCallback
        {
            public int lastBytes;
            public long lastFileSize;
            public byte[] lastData;

            public int Recv(long file_size, byte[] data, int bytes)
            {
                this.lastFileSize = file_size;
                this.lastData = data;
                this.lastBytes = bytes;
                return 0;
            }
        }

        [Fact]
        public void TestRecv()
        {
            var callback = new TestDownloadCallback();
            byte[] data = new byte[] { 1, 2, 3 };
            int result = callback.Recv(123L, data, data.Length);

            Assert.Equal(0, result);
            Assert.Equal(123L, callback.lastFileSize);
            Assert.Equal(data, callback.lastData);
            Assert.Equal(data.Length, callback.lastBytes);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/happyfish100_fastdfs-client-java/tests/original/download_callback_test.go
package original

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type TestDownloadCallback struct {
	LastBytes    int
	LastFileSize int64
	LastData     []byte
}

func (cb *TestDownloadCallback) Recv(fileSize int64, data []byte, bytes int) int {
	cb.LastFileSize = fileSize
	cb.LastData = data
	cb.LastBytes = bytes
	return 0
}

func TestRecv(t *testing.T) {
	cb := &TestDownloadCallback{}
	data := []byte{1, 2, 3}
	result := cb.Recv(123, data, len(data))
	assert.Equal(t, 0, result)
	assert.Equal(t, int64(123), cb.LastFileSize)
	assert.Equal(t, data, cb.LastData)
	assert.Equal(t, len(data), cb.LastBytes)
}